### ISSUE TITLE
conda-build issues on Unix

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/obspy-feedstock#1]
Seems the same issues described in this [comment](https://github.com/conda-forge/staged-recipes/pull/582#issuecomment-237832204) that plagued this recipe in staged-recipes are occurring in the feedstock. See this [CircleCI log](https://circleci.com/gh/conda-forge/obspy-feedstock/1) [Travis CI log](https://travis-ci.org/conda-forge/obspy-feedstock/jobs/150197883#L3077-L3109) for details. Snippet below. This appears to be caused by upstream issue ( https://github.com/conda/conda-build/issues/1167 ).

```
Traceback (most recent call last):
  File "/Users/travis/miniconda3/bin/conda-build", line 5, in <module>
    sys.exit(main())
  File "/Users/travis/miniconda3/lib/python3.5/site-packages/conda_build/main_build.py", line 151, in main
    args_func(args, p)
  File "/Users/travis/miniconda3/lib/python3.5/site-packages/conda_build/main_build.py", line 414, in args_func
    args.func(args, p)
  File "/Users/travis/miniconda3/lib/python3.5/site-packages/conda_build/main_build.py", line 357, in execute
    debug=args.debug)
  File "/Users/travis/miniconda3/lib/python3.5/site-packages/conda_build/build.py", line 632, in build
    preserve_egg_dir=bool(m.get_value('build/preserve_egg_dir')))
  File "/Users/travis/miniconda3/lib/python3.5/site-packages/conda_build/post.py", line 182, in post_process
    compile_missing_pyc(files)
  File "/Users/travis/miniconda3/lib/python3.5/site-packages/conda_build/post.py", line 174, in compile_missing_pyc
    call([python_exe, '-Wi', '-m', 'py_compile'] + compile_files, cwd=cwd)
  File "/Users/travis/miniconda3/lib/python3.5/subprocess.py", line 557, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/Users/travis/miniconda3/lib/python3.5/subprocess.py", line 947, in __init__
    restore_signals, start_new_session)
  File "/Users/travis/miniconda3/lib/python3.5/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/travis/miniconda3/envs/_build'
```

cc @msarahan @ocefpaf @megies @krischer

**Comment by megies** [comment]
Will be fixed by #2.

**Comment by megies** [comment]
Currently fixed by pinning to conda-build 1.21.7 and will be obsolete with conda-build 1.21.11.
